e2e: TestExamples dind-sidecar is consistently failing on main branch (and thus pull-request)

Any new pull-request is failing on that test. Something is broken in main.

```
2025-11-12T04:20:36.5504098Z     --- FAIL: TestExamples/v1/taskruns/dind-sidecar (15.31s)
```

--- a/examples/v1/taskruns/dind-sidecar.yaml
+++ b/examples/v1/taskruns/dind-sidecar.yaml
@@ -5,7 +5,7 @@ metadata:
 spec:
   taskSpec:
     steps:
-    - image: docker
+    - image: docker@sha256:0752ca4e936da012c173c119217c0f9599b3b191c1557e53206d5d06d2627580
       name: client
       env:
       # Connect to the sidecar over TCP, with TLS.

